lib/ts/recipe/emailpassword/validators.ts

```diff
@@ -0,0 +1,95 @@
+/* Copyright (c) 2020, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * defaultEmailValidator.
+ */
+
+export async function defaultEmailValidator(value: any): Promise<string | undefined> {
+    if (typeof value !== "string") {
+        return "Email must be of type string";
+    }
+
+    // eslint-disable-next-line no-useless-escape
+    const defaultEmailValidatorRegexp = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    // We check if the email syntax is correct
+    // As per https://github.com/supertokens/supertokens-auth-react/issues/5#issuecomment-709512438
+    // Regex from https://stackoverflow.com/a/46181/3867175
+
+    if (value.match(defaultEmailValidatorRegexp) === null) {
+        return "Email is invalid";
+    }
+
+    return undefined;
+}
+
+/*
+ * defaultPasswordValidator.
+ * min 8 characters.
+ * Contains lowercase, uppercase, and numbers.
+ */
+
+export async function defaultPasswordValidator(value: any): Promise<string | undefined> {
+    if (typeof value !== "string") {
+        return "Password must be of type string";
+    }
+
+    // length >= 8 && < 100
+    // must have a number and a character
+    // as per https://github.com/supertokens/supertokens-auth-react/issues/5#issuecomment-709512438
+
+    if (value.length < 8) {
+        return "Password must contain at least 8 characters, including a number";
+    }
+
+    if (value.length >= 100) {
+        return "Password's length must be lesser than 100 characters";
+    }
+
+    if (value.match(/^.*[A-Za-z]+.*$/) === null) {
+        return "Password must contain at least one alphabet";
+    }
+
+    if (value.match(/^.*[0-9]+.*$/) === null) {
+        return "Password must contain at least one number";
+    }
+
+    return undefined;
+}
+
+/*
+ * defaultLoginPasswordValidator.
+ * min 1 characters.
+ */
+
+export async function defaultLoginPasswordValidator(value: any): Promise<string | undefined> {
```

Comment:
On the backend, I do not have this kind of a function. For sign in, password validation, I simply return `undefined`. The core separately checks if the password is not an empty string and if it is, it throws the wrong credentials error.

Would you like to have the same behaviour here? Or shall I change the backend?

Reply:
I was under the impression that we decided to at least verify that there was one character but I didn't find any traces of this conversation. 

So I decided to check what big corps do:
- Facebook: no check for empty password
- Twitter: Disable the login button if empty password
- Instagram: Disable the login button if password is less than 6 characters

There is already a verification to make sure that all required fields are not empty:
![image](https://user-images.githubusercontent.com/5072452/98567473-40b31a00-22b0-11eb-877f-124b40da17d2.png)

Which makes the  `value.length === 0` check useless.

You don't need to implement the same thing in the backend, as you said, you can simply throw wrong credentials.